### ISSUE TITLE
Suspending functions support ?

### ISSUE THREAD

**Issue by robertoestivill** [holgerbrandl/kscript#119]
I'm currently experimenting with both `kscript` and `coroutines`. 

I have the following script that works as expected while using `runBlocking` and calling the suspending function `delay`

```
#!/usr/bin/env kscript
@file:DependsOn("org.jetbrains.kotlinx:kotlinx-coroutines-core:0.21")
import kotlinx.coroutines.experimental.*

runBlocking {
  println("start")
  delay(1000)
  println("end")
 }

$ kscript script.kts
start
end
```

However when I extract the lambda to it's own `suspend` function, the following compilation error happens:

```
#!/usr/bin/env kscript
@file:DependsOn("org.jetbrains.kotlinx:kotlinx-coroutines-core:0.21")
import kotlinx.coroutines.experimental.*

runBlocking {
  test()
}

suspend fun test(){
  println("start")
  delay(1000)
  println("end")
 }

$ kscript script.kts
[kscript] [ERROR] compilation of 'script.kts' failed
script.kts:11:2: error: modifier 'suspend' is not applicable to 'local function'
 suspend fun test(){
 ^
```

This is a simplified example of the issue I am running into with a bigger script.
Any ideas? Do I have to pass anything to enable coroutine support?

Thanks

**Comment by holgerbrandl** [comment]
Interesting question. It seems to be a kotlin bug. There are 2 related tickets

https://youtrack.jetbrains.com/issue/KT-20364 - There is no target version, but the ticket history suggests that it was fixed for the upcoming version of kotlin (v1.2.50 I guess).

https://youtrack.jetbrains.com/issue/KT-18508 - Open, but maybe it was not yet updated after the fix from above.

The following works fine for me as a workaround.
```
#!/usr/bin/env kscript
@file:DependsOn("org.jetbrains.kotlinx:kotlinx-coroutines-core:0.21")

import kotlinx.coroutines.experimental.*

runBlocking {
    Foo.test()
}

object Foo {
    suspend fun test() {
        println("start")
        delay(1000)
        println("end")
    }
}
```

According to https://kotlinlang.org/docs/diagnostics/experimental-coroutines.html you may need to  provide `@file:CompilerOpts("-Xcoroutines=enable")` (assuming that it's a compiler setting, otherwise it would be `file:KotlinOpts`). However the example from above works without, and I could not figure out a better example, where this setting is actually required.

**Comment by robertoestivill** [comment]
Thanks for the speedy and informative response. 
Will follow those bugs and use the workaround you provided.
Thanks :)
